### PR TITLE
[nrf fromlist] drivers: nrf5: fixes for the CSL Receiver

### PR DESCRIPTION
**Pull request by edmont**
Introduce fixes for making the CSL Receiver to properly schedule
the reception window.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

KRKNWK-11203

### CODE REVIEW

File: drivers/ieee802154/ieee802154_nrf5.c [hunk 1]
```diff
@@ -70,6 +70,7 @@ static struct nrf5_802154_data nrf5_data;
 #define DRX_SLOT_PH 0 /* Placeholder delayed reception window ID */
 #define DRX_SLOT_RX 1 /* Actual delayed reception window ID */
 #define PH_DURATION 10 /* Duration of the placeholder window, in microseconds  */
+#define DRX_ADJUST  700 /* Correction for the receive slot start */
```
Comment (hubertmis) [inline drivers/ieee802154/ieee802154_nrf5.c:73]: Why this adjustment is needed? Is there some offset between timer domains used by the radio driver and Zephyr?

Comment (e-rk) [reply drivers/ieee802154/ieee802154_nrf5.c:73]: I would also like to know where this value came from. Is it correct across all ranges of delayed RX duration and start times?

Comment (edmont) [reply drivers/ieee802154/ieee802154_nrf5.c:73]: The origin of the adjustment is still unknown, it was derived from trial and error testing with usual CSL configurations.

Comment (hubertmis) [reply drivers/ieee802154/ieee802154_nrf5.c:73]: I would clarify in the comment that it is a temporary workaround for a problem being investigated.

**Review by lmaciejonczyk — APPROVED** [review]

**Review by hubertmis — APPROVED** [review]

File: drivers/ieee802154/ieee802154_nrf5.c [hunk 2]
```diff
@@ -69,7 +69,12 @@ static struct nrf5_802154_data nrf5_data;
 
 #define DRX_SLOT_PH 0 /* Placeholder delayed reception window ID */
 #define DRX_SLOT_RX 1 /* Actual delayed reception window ID */
-#define PH_DURATION 10 /* Duration of the placeholder window, in microseconds  */
+#define PH_DURATION 10 /* Duration of the placeholder window, in microseconds */
+#define DRX_ADJUST                                                                                 \
+	800 /* When scheduling the actual delayed reception window an adjustment of 800 us is required
+		 * to match the CSL tranmission timing for unknown reasons. This is a temporary workaround
+		 * until the root cause is found.
+		 */
```
Comment (rlubos) [inline drivers/ieee802154/ieee802154_nrf5.c:77]: There are compliance issues with this comment, please see the CI results.

**Review by e-rk — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Comment by edmont** [comment]
Manifest PR is finally green.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**